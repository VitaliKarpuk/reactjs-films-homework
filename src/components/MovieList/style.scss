.movie-list{
  max-width: 100%;
  background-color: rgb(228, 228, 228);
  padding: 50px 30px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: flex-start;
}