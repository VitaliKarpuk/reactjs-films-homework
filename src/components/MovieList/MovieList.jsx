/* eslint-disable react/jsx-key */
import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import PropTypes from "prop-types";
import Film from "../Film/Film";
import requestFilmsByGenre from "../../modules/actions/requestFilmsByGenre";
import requesFilms from "../../modules/actions/requestFilms";
import { useParams } from "react-router";
import Loading from "../Loading/Loading";

import "./style.scss";

const MovieList = ({ films, genre, movieListView }) => {
  const { id: filmGenre, typeSearch } = useParams();
  const dispatch = useDispatch();
  const showListFilms = useSelector(state => state.showListFilms);
  const arrGenre = useSelector(state => state.genre);

  useEffect(() => {
    if (filmGenre && arrGenre) {
      dispatch(requestFilmsByGenre("popular", filmGenre, 1));
    } else {
      dispatch(requesFilms(typeSearch, 1));
    }
  }, [filmGenre, typeSearch, arrGenre]);

  const changeGenre = arr => {
    return arr.reduce((acc, item, index) => {
      genre.forEach(elem => {
        if (elem.id === item) {
          acc[index] = elem.name;
        }
      });
      return acc;
    }, []);
  };

  return (
    <section className="movie-list">
      {!showListFilms ? (
        <Loading />
      ) : (
        <div
          className={
            movieListView
              ? "movie-list__wrapper-items"
              : "movie-list__wrapper-items_active"
          }
        >
          {films.map(film => {
            return (
              <Film
                movieListView={movieListView}
                key={film.id}
                title={film.title}
                overview={film.overview}
                imgUrl={film.poster_path}
                rating={film.vote_average}
                genreFilmArray={film.genre_ids}
                genre={genre}
                changeGenre={changeGenre}
                id={film.id}
              />
            );
          })}
        </div>
      )}
    </section>
  );
};

MovieList.propTypes = {
  films: PropTypes.array,
  genre: PropTypes.array,
  movieListView: PropTypes.bool
};

MovieList.defaultProps = {
  films: [],
  genre: [],
  movieListView: true
};

export default MovieList;
