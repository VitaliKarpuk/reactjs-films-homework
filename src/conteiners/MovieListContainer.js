import { connect } from 'react-redux';
import MovieList from '../components/MovieList/MovieList';
import getFilms from '../modules/actions/getFilms';
import { getListFilms, getGenre } from '../modules/selectors/index';

const mapStateToPops = (state) => {
  return {
    films: getListFilms(state),
    genre: getGenre(state),
  }
}


const MovieListContainer = connect(mapStateToPops, dispatch => ({
  onGetilms: () => {
    dispatch(getFilms())
  }
}))(MovieList);

export default MovieListContainer;
