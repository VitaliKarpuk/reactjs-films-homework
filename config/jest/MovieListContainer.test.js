/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import cofigureMockStore from 'redux-mock-store';

import MovieListContainer from '../../src/conteiners/MovieListContainer';
import MovieList from '../../src/components/MovieList/MovieList';

const mockStore = cofigureMockStore();


describe('My Connected React-Redux Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      films: [{ id: 1, genre_ids: [] }],
      genre: [1, 2],
    });

    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <MovieListContainer />
      </Provider>
    );
  });


  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with given state from Redux store', () => {
    renderer.act(() => {
      <MovieList genreFilmArray={[{ id: 1 }]} films={[{ id: 1 }]} />
    });
    expect(store.dispatch).toHaveBeenCalledTimes(1);
  });
});