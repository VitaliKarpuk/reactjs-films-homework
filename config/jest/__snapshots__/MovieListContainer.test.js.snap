// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<section
  className="movie-list"
>
  <div
    className="movie-list__item"
  >
    <div
      className="movie-list__item_img"
    >
      
    </div>
    <div
      className="movie-list__rating"
    >
      <h3>
        
      </h3>
      <span>
        0
      </span>
    </div>
    <p
      className="movie-list__genre"
    />
    <div
      className="movie-hover"
    >
      <div
        className="movie-hover__icon"
        onClick={[Function]}
      />
      <h3>
        Watch Now
      </h3>
      <button
        className="btn__info"
        onClick={[Function]}
      >
        View Info
      </button>
    </div>
  </div>
</section>
`;
